lib/facility.rb

```diff
@@ -1,14 +1,97 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :collected_fees, :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+   
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def log_vehicle(vehicle)
+    @registered_vehicles << vehicle
+  end
+
+  def update_registration(vehicle)
+   
+
+    vehicle.registration_date = Date.today
+  end
+
+  def plate_service(vehicle)
+    #if it is antique
+    if vehicle.antique?
+      then @collected_fees += 25
```

Comment:
Collecting fees could be done in a small helper method to better promote SRP. Every function should do just one thing, really well - even if it means calling on a lot of helper functions